zerver/views/message_fetch.py

```diff
@@ -677,19 +715,21 @@ def get_base_query_for_search(user_profile: UserProfile,
     inner_msg_id_col = column("message_id")
     return (query, inner_msg_id_col)
 
-def add_narrow_conditions(user_profile: UserProfile,
+def add_narrow_conditions(user_profile: Optional[UserProfile],
                           inner_msg_id_col: ColumnElement,
                           query: Query,
-                          narrow: OptionalNarrowListT) -> Tuple[Query, bool]:
+                          narrow: OptionalNarrowListT,
+                          is_web_public_query: bool,
+                          realm: Realm) -> Tuple[Query, bool]:
     is_search = False  # for now
 
-    if narrow is None:
-        return (query, is_search)
-
```

Comment:
What's the purpose of moving this block lower down? It looks like in the `narrow is None` case we'll just do the extra work of `builder = NarrowBuilder(user_profile, inner_msg_id_col, realm, is_web_public_query)` but end up returning the same thing as before anyway?

Reply:
Thanks for catching this, I was initially thinking of allowing narrow=None in web-public queries. I will correct this.